File: ruby/rot26.rb
# Ruby rot26 implementation
# by ArmoredStone
# 24.01.2025

def rotate_char(input_char, mode)
    shift = 26
    if mode == 'left'
        return (input_char.ord - shift).chr
    elsif mode == 'right'
        return (input_char.ord + shift).chr
    else
        puts "Usage: ruby rot26.rb [left|right] [string]"
        exit
    end
end

def rotate_string(input_string, mode)
    return input_string.chars.map { |c| rotate_char(c, mode) }.join
end

if ARGV.length != 2
    puts "Usage: ruby rot26.rb [left|right] [string]"
    exit
end

mode = ARGV[0]
string = ARGV[1]

unless ['left', 'right'].include?(mode)
    puts "Usage: ruby rot26.rb [left|right] [string]"
    exit
end

if string.nil?
    puts "Usage: ruby rot26.rb [left|right] [string]"
    exit
end

result = rotate_string(string, mode)
puts result

File: python/rot26.py
# Python rot26 implementation
# by ArmoredStone
# 24.01.2025

import sys

def rotate_string(input_str, mode):
    result = ""
    shift = 26
    for k in input_str:
        if mode == 'left':
            result += chr(ord(k) - shift)
        elif mode == 'right':
            result += chr(ord(k) + shift)
        else:
            print("Usage: python rot26.py [left|right] <string>")
            sys.exit(1)
    return result

def main():
    if len(sys.argv) != 3:
        print("Usage: python rot26.py [left|right] <string>")
        sys.exit(1)
    
    mode = sys.argv[1]
    input_string = str(sys.argv[2])

    if mode not in ['left', 'right']:
        print("Mode must be either 'left' or 'right'")
        sys.exit(1)

    rotated_string = rotate_string(input_string, mode)
    print(rotated_string)

if __name__ == "__main__":
    main()
